src/gtk/toga_gtk/widgets/canvas.py

```diff
@@ -134,9 +134,11 @@ def fill(self, color, fill_rule, preserve, draw_context, *args, **kwargs):
         else:
             draw_context.fill()
 
-    def stroke(self, color, line_width, draw_context, *args, **kwargs):
+    def stroke(self, color, line_width, line_dash, draw_context, *args, **kwargs):
         self.apply_color(color, draw_context)
         draw_context.set_line_width(line_width)
+        if line_dash is not None:
+            draw_context.set_dash(line_dash, 1)
```

Comment:
Did you set the offset to 1 in order to match the cocoa implementation?

Reply:
I did not implement the dash offset for cocoa, although it would be easy to add. I would recommend setting the offset to zero and adding to the api to add the dash offset

Reply:
No, I just realized I was looking at the cairo_set_dash function and not the pycairo set_dash function. I thought the 1 was to keep the dash symmetric, which I now realize is silly.